CMakePresets.json

```diff
@@ -29,6 +29,18 @@
         "CMAKE_BUILD_TYPE": "RelWithDebInfo"
       }
     },
+    {
+      "name": "Server-PostgreSQL",
```

Comment:
this file currently exactly mirrors the cmake build types, I don't think we need more